content/en/blog/2024/otel-generative-ai/index.md

```diff
@@ -0,0 +1,147 @@
+---
+title: OpenTelemetry for Generative AI
+linkTitle: OpenTelemetry for Generative AI
+date: 2024-11-09
+author: >-
+  [Drew Robbins](https://github.com/drewby) (Microsoft),  [Liudmila
+  Molkova](https://github.com/lmolkova) (Microsoft)
+---
+
+As organizations increasingly adopt Large Language Models (LLMs) and other
+generative AI technologies, ensuring reliable performance, efficiency, and
+safety is essential to meet user expectations, optimize resource costs, and
+safeguard against unintended outputs. Effective observability for AI operations,
+behaviors, and outcomes can help meet these goals. OpenTelemetry is being
+enhanced to support these needs specifically for generative AI.
+
+Two primary assets are in development to make this possible: **Semantic
+Conventions** and an **Instrumentation Library**.
+
+**Semantic Conventions** establish standardized guidelines for how telemetry
+data is structured and collected across platforms, defining inputs, outputs, and
+operational details. For generative AI, these conventions streamline monitoring,
+troubleshooting, and optimizing AI models by standardizing attributes such as
+model parameters, response metadata, and token usage. This consistency supports
+better observability across tools, environments, and APIs, helping organizations
+track performance, cost, and safety with ease.
+
+The **Instrumentation Library** is being developed within the OpenTelemetry
+Python Contrib project to automate telemetry collection for generative AI
+applications. The first release is a Python library, given Python’s widespread
+use in AI development. Designed to integrate seamlessly with OpenAI’s API, this
+library captures spans and events, gathering essential data like model
+inputs, response metadata, and token usage in a structured format.
+
+## Key Signals for Generative AI
+
+The
+[Semantic Conventions for Generative AI](https://github.com/open-telemetry/semantic-conventions/tree/v1.28.0/docs/gen-ai)
+focus on capturing insights into AI model behavior through three primary
+signals: Spans, Metrics, and Events.
```

Comment:
```suggestion
signals: [Traces](/docs/concepts/signals/traces/), [Metrics](/docs/concepts/signals/metrics/), and [Logs](/docs/concepts/signals/logs/).
```

Re events: I am fine with that as well, but people will stumble upon that and ask themselves "why events and not logs", so you would need to add some explanation and links about it. I know that this is a highly debated subject within the community but this blog post will be consumed by a lot of people that do not know all the things happening within OTel so we need to be cautious to put words out there without adding some more context.